File: src/templates/java/lib_io.java
// ═══════════════════════════════════════════════════════════════════════════
// Java 25 - Librería de I/O
// ═══════════════════════════════════════════════════════════════════════════
// Esta librería proporciona operaciones de entrada/salida mejoradas
// que pueden ser heredadas por otros nodos usando ch("lib_io")

import java.io.*;
import java.nio.file.*;
import java.util.*;
import java.util.stream.*;

public class IOUtils {
    // Leer archivo completo como String
    public static String readFile(String ruta) throws IOException {
        return Files.readString(Path.of(ruta));
    }
    
    // Escribir String a archivo
    public static void writeFile(String ruta, String contenido) throws IOException {
        Files.writeString(Path.of(ruta), contenido);
    }
    
    // Leer archivo línea por línea
    public static List<String> readLines(String ruta) throws IOException {
        return Files.readAllLines(Path.of(ruta));
    }
    
    // Escribir líneas a archivo
    public static void writeLines(String ruta, List<String> lineas) throws IOException {
        Files.write(Path.of(ruta), lineas);
    }
    
    // Leer archivo como Stream de líneas
    public static Stream<String> readLinesStream(String ruta) throws IOException {
        return Files.lines(Path.of(ruta));
    }
    
    // Verificar si archivo existe
    public static boolean fileExists(String ruta) {
        return Files.exists(Path.of(ruta));
    }
    
    // Crear directorio si no existe
    public static void createDirectory(String ruta) throws IOException {
        Files.createDirectories(Path.of(ruta));
    }
    
    // Listar archivos en directorio
    public static List<String> listFiles(String directorio) throws IOException {
        return Files.list(Path.of(directorio))
            .map(p -> p.getFileName().toString())
            .collect(Collectors.toList());
    }
    
    // Copiar archivo
    public static void copyFile(String origen, String destino) throws IOException {
        Files.copy(Path.of(origen), Path.of(destino), StandardCopyOption.REPLACE_EXISTING);
    }
    
    // Leer propiedades desde archivo
    public static Properties loadProperties(String ruta) throws IOException {
        Properties props = new Properties();
        try (FileInputStream fis = new FileInputStream(ruta)) {
            props.load(fis);
        }
        return props;
    }
    
    // Guardar propiedades a archivo
    public static void saveProperties(Properties props, String ruta) throws IOException {
        try (FileOutputStream fos = new FileOutputStream(ruta)) {
            props.store(fos, "Propiedades guardadas");
        }
    }
    
    // Ejemplo de uso
    public static void main(String[] args) {
        try {
            // Crear archivo de prueba
            String ruta = "test.txt";
            writeFile(ruta, """
                Línea 1
                Línea 2
                Línea 3
                """);
            
            // Leer archivo
            String contenido = readFile(ruta);
            System.out.println("Contenido:\n" + contenido);
            
            // Leer líneas
            List<String> lineas = readLines(ruta);
            System.out.println("Líneas: " + lineas);
            
            // Limpiar
            Files.deleteIfExists(Path.of(ruta));
        } catch (IOException e) {
            System.err.println("Error de I/O: " + e.getMessage());
        }
    }
}

File: src/templates/rust/lib_io.rs
/* ═══════════════════════════════════════════════════════════════════════════
 * RUST IO - Utilidades de entrada/salida
 * ═══════════════════════════════════════════════════════════════════════════
 * Archivo: lib_io.rs
 * Descripción: Funciones de I/O reutilizables para Rust
 * 
 * USO: Este módulo puede ser heredado por otros nodos para usar estas funciones
 * Ejemplo de herencia: Conecta este nodo a otro para acceder a io con ch()
 * ═══════════════════════════════════════════════════════════════════════════
 */

use std::fs;
use std::io::{self, Read, Write};
use std::path::Path;

// ═══════════════════════════════════════════════════════════════════════════
// LECTURA DE ARCHIVOS
// ═══════════════════════════════════════════════════════════════════════════

/// Leer archivo completo como String
pub fn read_file_string(path: &str) -> Result<String, io::Error> {
    fs::read_to_string(path)
}

/// Leer archivo completo como bytes
pub fn read_file_bytes(path: &str) -> Result<Vec<u8>, io::Error> {
    fs::read(path)
}

/// Verificar si archivo existe
pub fn file_exists(path: &str) -> bool {
    Path::new(path).exists()
}

/// Obtener tamaño de archivo
pub fn file_size(path: &str) -> Result<u64, io::Error> {
    let metadata = fs::metadata(path)?;
    Ok(metadata.len())
}

// ═══════════════════════════════════════════════════════════════════════════
// ESCRITURA DE ARCHIVOS
// ═══════════════════════════════════════════════════════════════════════════

/// Escribir string a archivo
pub fn write_file_string(path: &str, content: &str) -> Result<(), io::Error> {
    fs::write(path, content)
}

/// Escribir bytes a archivo
pub fn write_file_bytes(path: &str, content: &[u8]) -> Result<(), io::Error> {
    fs::write(path, content)
}

/// Crear directorio (con padres si no existen)
pub fn create_dir_all(path: &str) -> Result<(), io::Error> {
    fs::create_dir_all(path)
}

/// Eliminar archivo
pub fn remove_file(path: &str) -> Result<(), io::Error> {
    fs::remove_file(path)
}

/// Eliminar directorio (vacío)
pub fn remove_dir(path: &str) -> Result<(), io::Error> {
    fs::remove_dir(path)
}

// ═══════════════════════════════════════════════════════════════════════════
// UTILIDADES DE PATH
// ═══════════════════════════════════════════════════════════════════════════

/// Obtener extensión de archivo
pub fn get_extension(path: &str) -> Option<String> {
    Path::new(path)
        .extension()
        .and_then(|ext| ext.to_str())
        .map(|s| s.to_string())
}

/// Obtener nombre de archivo (sin path)
pub fn get_filename(path: &str) -> Option<String> {
    Path::new(path)
        .file_name()
        .and_then(|name| name.to_str())
        .map(|s| s.to_string())
}

/// Obtener directorio padre
pub fn get_parent(path: &str) -> Option<String> {
    Path::new(path)
        .parent()
        .and_then(|p| p.to_str())
        .map(|s| s.to_string())
}

/// Unir paths
pub fn join_paths(path1: &str, path2: &str) -> String {
    Path::new(path1)
        .join(path2)
        .to_string_lossy()
        .to_string()
}

// ═══════════════════════════════════════════════════════════════════════════
// ENTRADA ESTÁNDAR
// ═══════════════════════════════════════════════════════════════════════════

/// Leer línea desde stdin
pub fn read_line() -> Result<String, io::Error> {
    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer)?;
    Ok(buffer.trim().to_string())
}

/// Escribir a stdout
pub fn print(text: &str) {
    print!("{}", text);
    io::stdout().flush().ok();
}

/// Escribir línea a stdout
pub fn println(text: &str) {
    println!("{}", text);
}

